### PR TITLE
Modularize automation

### PR DESCRIPTION
**Pull request by christophmeissner**
* splits `makemessages.sh` into multiple single scripts in `scripts` dir
* split workflow into multiple independent jobs

### CODE REVIEW

File: checkmessages.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+#set -e
```
Comment (pitpalme) [inline checkmessages.sh:2]: Why deactivated?

Comment (christophmeissner) [reply checkmessages.sh:2]: Checks are independent from each other and failing one of them does not necessary require to abort all subsequent checks.

File: scripts/checkdiffs.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+
+echo
+echo "================================================================================"
+echo "Checking for changes in .po files..."
+echo "--------------------------------------------------------------------------------"
+diff_cmd="git diff"
+diff_args=(--ignore-matching-lines=POT-Creation-Date --exit-code -- '***.po')
```
Comment (pitpalme) [inline scripts/checkdiffs.sh:8]:
```suggestion
diff_cmd="git"
diff_args=(diff --ignore-matching-lines=POT-Creation-Date --exit-code -- ':/:locale/**/*.po')
```

File: scripts/checkfuzzy.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+echo
+echo "================================================================================"
+echo "Checking for fuzzy translations..."
+echo "--------------------------------------------------------------------------------"
+grep_cmd="grep"
+grep_args=(-r '#, fuzzy' locale/)
```
Comment (pitpalme) [inline scripts/checkfuzzy.sh:8]:
```suggestion
grep_cmd="grep"
grep_args=(-rE '^#[,[:space:]]+fuzzy' locale/)
```

File: scripts/checkfuzzy.sh [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+echo "${grep_cmd} ${grep_args[*]}"
+${grep_cmd} "${grep_args[@]}"
+grep_result=$?
+
+if [ $grep_result -ne 0 ]; then
+    echo "No fuzzy translations found."
+    has_fuzzy=0
+else
+    has_fuzzy=1
+fi
```
Comment (pitpalme) [inline scripts/checkfuzzy.sh:18]:
```suggestion
[ $grep_result -ne 0 ]
has_fuzzy=$?

if [ $has_fuzzy -eq 0 ]; then
    echo "No fuzzy translations found."
fi
```

**Review by pitpalme — CHANGES_REQUESTED** [review]

**Review by pitpalme — APPROVED** [review]

**Pull request merged**